resource/resource_graph.go

```diff
@@ -118,6 +119,46 @@ func (g *Graph) Clone() *Graph {
 	return g.clone()
 }
 
+// Dot exports the resource graph as a DOT file for vizualization
+func (g *Graph) Dot() ([]byte, error) {
```

Comment:
I was kind of partial to your original `Export` function name. Or `ExportDot` to describe the format being exported to.